### PR TITLE
fix: correct multi-env example_id collision in /all/ metrics

### PR DESCRIPTION
**Pull request by jfeldstein**
# Context

When multiple environments each auto-assign integer `example_ids` from `range(len(dataset))` — the default for every built-in environment — `groupby("example_id")` silently merges rollouts from different problems across environments into invalid groups. 

# Issue

This corrupts `solve_all/all`, `solve_none/all`, `effective_batch_size/all`, and `all seq_len/reward/all/*` metrics logged to W&B in any multi-environment training run. 

# Impact

If anyone is manually using `solve_all/all` or `reward/all/mean` as a stopping signal or to make decisions about hyperparameters mid-run, those decisions could be wrong. But **the automated training loop itself is clean**.

# Fix

- Extract compute_solve_rates into a new, minimal metrics.py module (no heavy deps, directly unit-testable without torch/transformers)
- Group by ["env_name", "example_id"] instead of "example_id" alone in both compute_solve_rates and the by_example groupby
- Add regression tests covering single-env baseline, the collision scenario, and three-env mixed cases

The per-env metrics path (env_df.groupby("example_id")) was already correct because env_df is pre-filtered to a single environment.

---

> [!NOTE]
> **Medium Risk**
> Changes how global training metrics are grouped and computed, which can alter W&B dashboards and any human-in-the-loop stopping heuristics, but does not affect the training data sent to the trainer.
> 
> **Overview**
> Fixes corrupted `/all/` metrics in multi-environment training runs by grouping rollouts by `("env_name", "example_id")` instead of `"example_id"` alone, preventing cross-environment ID collisions from skewing logged `reward/*`, `seq_len/*`, and solve-rate metrics.
> 
> Extracts `compute_solve_rates` into a new lightweight `orchestrator/metrics.py` and adds unit tests covering single-env baselines and multi-env collision regression scenarios.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit bf3d67f9c3075788ec885648beaf812c2c087232. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/prime_rl/orchestrator/metrics.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+Kept in a separate module with no heavy imports so they can be unit-tested
+without pulling in torch / transformers / verifiers.
+"""
+
+from __future__ import annotations
+
+import pandas as pd
+
+
+def compute_solve_rates(
```
Comment (mikasenghaas) [inline src/prime_rl/orchestrator/metrics.py:12]: lets inline this for the time being pls

File: tests/unit/orchestrator/test_metrics.py
Comment (mikasenghaas) [inline tests/unit/orchestrator/test_metrics.py:1]: no need for tests here

**Review by mikasenghaas — COMMENTED** [review]
thx for the pr! pls address the 2 comments and then we will get this in asap

File: tests/unit/orchestrator/test_metrics.py
Comment (jfeldstein) [reply tests/unit/orchestrator/test_metrics.py:1]: Why no tests for this?

Happy to clean up, but they do cover a bug in code that previously lacked coverage
